packages/core-test-kit/src/inline-expectation.ts

```diff
@@ -0,0 +1,131 @@
+        const prefix = msg ? msg + `: ` : ``;
+        if (rule.selector !== expectedSelector) {
+            errors.push(`${prefix}expected ${rule.selector} to transform to ${expectedSelector}`);
+        }
+        if (declarationCheck === `full`) {
+            const actualDecl = rule.nodes.map((x) => x.toString()).join(`;`);
+            const expectedDecl = expectedDeclarations
+                .map(([prop, value]) => `${prop}: ${value}`)
+                .join(`;`);
+            if (actualDecl !== expectedDecl) {
+                errors.push(
+                    `${prefix}expected ${rule.selector} to have declaration {${expectedDecl}}, but got {${actualDecl}}`
+                );
+            }
+        }
+    });
+    // report errors
+    if (errors.length) {
+        throw new Error('\n' + errors.join('\n'));
+    }
+    if (expectedTestsCount !== checks.length) {
+        throw new Error(
+            `Expected ${expectedTestsCount} checks to run but there was ${checks.length}`
+        );
+    }
+}
+
+function createRuleCheck(rule: postcss.Rule, expectInput: string, errors: string[]): RuleCheck | undefined {
+    const { msg, ruleIndex, expectedSelector, expectedBody } = expectInput.match(
+        /(?<msg>\(.*\))*(\[(?<ruleIndex>\d+)\])*(?<expectedSelector>[^{}]*)\s*(?<expectedBody>.*)/s
+    )!.groups!;
+    const targetRule = ruleIndex ? getNextMixinRule(rule, Number(ruleIndex)) : rule;
+    if (!targetRule) {
+        errors.push(`cannot locate mixed-in rule for "${expectInput}"`);
+        return;
+    }
+    const expectedDeclarations: RuleCheck[`expectedDeclarations`] = [];
+    const declsInput = expectedBody.trim().match(/^{(.*)}$/s);
+    const declarationCheck: RuleCheck[`declarationCheck`] = declsInput ? `full` : `none`;
+    if (declsInput && declsInput[1]?.includes(`:`)) {
+        for (const decl of declsInput[1].split(`;`)) {
+            if (decl.trim() !== ``) {
+                const [prop, value] = decl.split(':');
+                if (prop && value) {
+                    expectedDeclarations.push([prop.trim(), value.trim()]);
+                } else {
+                    errors.push(`error in expectation "${decl}" of "${expectInput}"`);
+                }
+            }
+        }
+    }
+    return {
+        msg,
+        rule: targetRule,
+        expectedSelector: expectedSelector.trim(),
+        expectedDeclarations,
+        declarationCheck,
+    };
+}
+
+function getNextMixinRule(currentRule: postcss.Rule, count: number): postcss.Rule | undefined {
+    while (currentRule && count > 0) {
+        const next: postcss.ChildNode | undefined = currentRule.next();
+        // next must be a rule sense mixin can only add rules
+        if (next?.type === `rule`) {
+            currentRule = next;
+            count--;
+        } else {
+            return;
+        }
+    }
+    return currentRule && count === 0 ? currentRule : undefined;
+}
```

Comment:
missing newline. do you use prettier?

Reply:
Used it at some point, might have forgotten to run before last commit